icontrol/session.py

```diff
@@ -86,11 +86,22 @@ def _validate_icruri(base_uri):
     scheme, netloc, path, _, _ = urlsplit(base_uri)
     if scheme != 'https':
         raise InvalidScheme(scheme)
-    if not path.startswith('/mgmt/tm/'):
-        error_message = "The path must start with '/mgmt/tm/'!!  But it's:" +\
+
+    # This can be iWorkflow or BIG-IQ
+    if path.startswith('/mgmt/shared/'):
+        return _validate_prefix_collections(path[13:])
+
+    # This can also be in iWorkflow or BIG-IQ
+    if path.startswith('/mgmt/cm/'):
+        return _validate_prefix_collections(path[9:])
+
+    # Most of the time this is BIG-IP
+    if path.startswith('/mgmt/tm/'):
+        return _validate_prefix_collections(path[9:])
+    else:
+        error_message = "The path must start with '/mgmt/tm/'!  But it's:" +\
```

Comment:
my original method of doing this did indeed remove that string slice and just burped out the url the user sent, but it required changing the tests to make that work and then i got lazy. I think you're right that i need to go add that back though. Lemme also address the if/else comment